src/sentry/replays/lib/storage.py

```diff
@@ -0,0 +1,161 @@
+        """Remove a blob from remote storage."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def get(self, segment: RecordingSegmentStorageMeta) -> bytes:
+        """Return blob from remote storage."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def set(self, segment: RecordingSegmentStorageMeta, value: bytes) -> None:
+        """Set blob in remote storage."""
+        raise NotImplementedError
+
+
+class FilestoreBlob(Blob):
+    """Filestore service driver blob manager."""
+
+    def delete(self, segment: RecordingSegmentStorageMeta) -> None:
+        file = File.objects.get(pk=segment.file_id)
+        file.delete()
+
+    @metrics.wraps("replays.lib.storage.FilestoreBlob.get")
+    def get(self, segment: RecordingSegmentStorageMeta) -> bytes:
+        file = File.objects.get(pk=segment.file_id)
+        return file.getfile().read()
+
+    @metrics.wraps("replays.lib.storage.FilestoreBlob.set")
+    def set(self, segment: RecordingSegmentStorageMeta, value: bytes) -> None:
+        file = File.objects.create(name=make_filename(segment), type="replay.recording")
+        file.putfile(BytesIO(value), blob_size=settings.SENTRY_ATTACHMENT_BLOB_SIZE)
+
+        try:
+            segment = ReplayRecordingSegment.objects.create(
+                project_id=segment.project_id,
+                replay_id=segment.replay_id,
+                segment_id=segment.segment_id,
+                file_id=file.id,
+                size=len(value),
+            )
+        except IntegrityError:
+            logger.warning(
+                "Recording-segment has already been processed.",
+                extra={
+                    "replay_id": segment.replay_id,
+                    "project_id": segment.project_id,
+                    "segment_id": segment.segment_id,
+                },
+            )
+
+            # Delete the file from remote storage since it is un-retrievable.
+            file.delete()
+
+
+class StorageBlob(Blob):
+    """Storage service driver blob manager.
+
+    This driver does not have managed TTLs.  To enable TTLs you will need to enable it on your
+    bucket.  Keys are prefixed by their TTL.  Those TTLs are 30, 60, 90.  Measured in days.
+    """
+
+    def delete(self, segment: RecordingSegmentStorageMeta) -> None:
+        storage = get_storage(self._make_storage_options())
+        storage.delete(self.make_key(segment))
+
+    @metrics.wraps("replays.lib.storage.StorageBlob.get")
+    def get(self, segment: RecordingSegmentStorageMeta) -> bytes:
+        cache_key = make_filename(segment)
+
+        # Return eagerly from cache if the value exists.
+        cache_value = replay_cache.get(cache_key, raw=True)
+        if cache_value:
+            return cache_value
+
+        try:
+            storage = get_storage(self._make_storage_options())
+            blob = storage.open(self.make_key(segment))
+            result = blob.read()
+            blob.close()
+        except Exception:
```

Comment:
is there a more specific error we can `except` on than `Exception`?

Reply:
adding note from slack convo: it's hard as the underlying FileStore mechanism error isn't known.

Reply:
nit not blocking merge: should we log here for this exception?

Reply:
Logging the exception 👍